lib/tower/igniter.ex

```diff
@@ -1,14 +1,82 @@
 if Code.ensure_loaded?(Igniter) do
   defmodule Tower.Igniter do
-    def add_reporter_to_config(igniter, reporter) do
-      igniter
-      |> Igniter.Project.Config.configure(
+    @runtime_file_path "config/runtime.exs"
+    @default_runtime_file_content """
+    import Config
+    """
+
+    def reporters_list_append(igniter, module) do
+      Igniter.Project.Config.configure(
+        igniter,
         "config.exs",
         :tower,
         [:reporters],
-        [reporter],
-        updater: &Igniter.Code.List.append_new_to_list(&1, reporter)
+        [module],
+        updater: &Igniter.Code.List.append_new_to_list(&1, module)
+      )
+    end
+
+    def runtime_configure_reporter(igniter, application, items, opts \\ []) do
+      env = opts[:env]
+
+      igniter
+      |> Igniter.include_or_create_file(@runtime_file_path, @default_runtime_file_content)
+      |> Igniter.update_elixir_file(
+        @runtime_file_path,
+        fn zipper ->
+          if Igniter.Project.Config.configures_root_key?(zipper, application) do
+            {:ok, zipper}
+          else
+            if env do
+              zipper
+              |> Igniter.Code.Common.move_to_cursor_match_in_scope(env_config_patterns(env))
+            else
+              {:ok, zipper}
+            end
+            |> case do
+              {:ok, zipper} ->
+                if Igniter.Project.Config.configures_root_key?(zipper, application) do
+                  {:ok, zipper}
+                else
+                  Igniter.Code.Common.add_code(zipper, config_block(application, items))
+                end
+
+              :error ->
+                Igniter.Code.Common.add_code(zipper, env_config_block(env, application, items))
+            end
+          end
+        end
```

Comment:
This function does a mix of what `Igniter.configure_runtime_env` and `Igniter.configure_group` do I think...

Maybe there's room for an `Igniter.configure_runtime_env_group` or something like that...